src/blocks/maildir.rs

```diff
@@ -77,6 +77,21 @@ impl ConfigBlock for Maildir {
         shared_config: SharedConfig,
         _tx_update_request: Sender<Task>,
     ) -> Result<Self> {
+        let expanded_inboxes = block_config
+            .inboxes
+            .iter()
+            .map(|inbox| {
+                shellexpand::full(inbox)
+                    .map_err(|e| {
+                        ConfigurationError(
+                            "maildir".to_string(),
+                            format!("Failed to expand inbox path {}: {}", inbox, e),
+                        )
+                    })
+                    .unwrap()
```

Comment:
Or, alternatively (perhaps preferably), this can be done in-place:
```rust
for inbox in &mut block_config.inboxes {
    *inbox = shellexpand::full(inbox).map_err(...)?;
}
```